pkg/agent/datamodel/sig_config.go

```diff
@@ -12,6 +16,136 @@ const (
 	AzurePublicCloudSigSubscription string = "109a5e88-712a-48ae-9078-9ca8b3c81345" // AKS VHD
 )
 
+//nolint:gochecknoglobals
+var (
+	CachedFromComponentContainerImages = make(map[string]ContainerImage)
+	CachedFromComponentDownloadedFiles = make(map[string]DownloadFiles)
+	CachedFromManifest                 = &Manifest{}
+)
+
+//nolint:gochecknoinits
+func init() {
+	manifest := CacheManifest()
+	processManifest(manifest)
+
+	components := CacheComponents()
+	processComponents(components)
+}
+
+func CacheManifest() Manifest {
+	_, filename, _, _ := runtime.Caller(0)
+	manifestFilePath := "../../../parts/linux/cloud-init/artifacts/manifest.json"
+	return getCachedVersionsFromManifestJSON(path.Join(path.Dir(filename), manifestFilePath))
+}
+
+func CacheComponents() Components {
+	_, filename, _, _ := runtime.Caller(0)
+	componentsFilePath := "../../../vhdbuilder/packer/components.json"
+	return getCachedVersionsFromComponentsJSON(path.Join(path.Dir(filename), componentsFilePath))
+}
+
+func processManifest(manifest Manifest) {
+	CachedFromManifest.Kubernetes = manifest.Kubernetes
+	CachedFromManifest.Runc = manifest.Runc
+	CachedFromManifest.Containerd = manifest.Containerd
+	CachedFromManifest.NvidiaContainerRuntime = manifest.NvidiaContainerRuntime
+	CachedFromManifest.NvidiaDrivers = manifest.NvidiaDrivers
+}
+
+func processComponents(components Components) {
```

Comment:
I'd prefer this func return an error and then we panic on said errors directly within the `init()`